src/transaction.ts

```diff
@@ -253,6 +255,8 @@ export class Transaction implements TransactionStorageStructure {
       throw Error('firstRound must be a positive number');
     if (!Number.isSafeInteger(txn.lastRound) || txn.lastRound < 0)
       throw Error('lastRound must be a positive number');
+    if (txn.extraPages < 0 || txn.extraPages > 3)
```

Comment:
What was the issue with `Number.isInteger`?

Reply:
It somehow causes all the tests to fail with extra pages parameter to fail. Odd because they are most definitely integers

Reply:
It's because I didn't check for undefined first 🤦‍♂️

Reply:
```suggestion
    if (txn.extraPages != null && (!Number.isInteger(txn.extraPages) || txn.extraPages < 0 || txn.extraPages > 3))
```